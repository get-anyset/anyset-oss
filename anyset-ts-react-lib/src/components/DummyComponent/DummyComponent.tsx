import React from 'react';

export interface DummyComponentProps {
  title?: string;
}

export const DummyComponent: React.FC<DummyComponentProps> = ({ title = 'Dummy Component' }) => {
  return (
    <div>
      <h1>{title}</h1>
      <p>This is a dummy component for testing purposes.</p>
    </div>
  );
};
