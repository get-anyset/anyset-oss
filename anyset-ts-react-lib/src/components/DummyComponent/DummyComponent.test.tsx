import { render } from '@testing-library/react';

import { DummyComponent } from './DummyComponent';

describe('DummyComponent', () => {
  it('renders without crashing', () => {
    const app = render(<DummyComponent />);
    const element = app.getByText('Dummy Component');
    expect(element).toBeDefined();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<DummyComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
