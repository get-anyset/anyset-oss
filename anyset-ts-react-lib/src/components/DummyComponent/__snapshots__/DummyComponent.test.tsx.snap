// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DummyComponent matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Dummy Component
    </h1>
    <p>
      This is a dummy component for testing purposes.
    </p>
  </div>
</DocumentFragment>
`;
